### PR TITLE
fix(handlebars): Use quotes from config in AttrNode

### PR DESCRIPTION
**Pull request by dcyriller**
Closes https://github.com/prettier/prettier/issues/6376

- [x] I’ve added tests to confirm my change works.
- [x] (If the change is user-facing) I’ve added my changes to the `CHANGELOG.unreleased.md` file following the template.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by dcyriller** [comment]
@duailibe it’s rebased now! (I was on vacations.)

**Comment by dcyriller** [comment]
> /cc @dcyriller need rebase again, thanks!

done, let me add the test

**Comment by dcyriller** [comment]
I trembled a bit, but it has worked out great: https://github.com/prettier/prettier/pull/6377/commits/8ac337d7a05a3a393bdb818cb1ca75c80d0c956b 😄

File: tests/handlebars-attr-node/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+parsers: ["glimmer"]
+printWidth: 80
+singleQuote: true
+                                                                                | printWidth
+=====================================input======================================
+<div title='My "title"'></div>
+<div title="My other 'title'"></div>
+
+=====================================output=====================================
+<div title='My "title"'></div><div title="My other 'title'"></div>
```
Comment (evilebottnawi) [inline tests/handlebars-attr-node/__snapshots__/jsfmt.spec.js.snap]: I think we need quotes here like do prettier for js

Comment (dcyriller) [reply tests/handlebars-attr-node/__snapshots__/jsfmt.spec.js.snap]: You mean, we should have for the second div?
```
<div title='My other \'title\''></div>
```

On master, hbs printer produces the following output (the two divs wrong, the first being very wrong):
```
<div title="My "title""></div><div title="My other 'title'"></div>
```

Comment (evilebottnawi) [reply tests/handlebars-attr-node/__snapshots__/jsfmt.spec.js.snap]: hm, all fine, my mistake

File: tests/handlebars-basics/__snapshots__/jsfmt.spec.js.snap [hunk 2]
```diff
@@ -51,12 +51,12 @@ singleQuote: true
 <div class="{{hello}} {{world}}"></div>
 
 =====================================output=====================================
-<script id="entry-template" type="text/x-handlebars-template">
-  <div class="entry">
+<script id='entry-template' type='text/x-handlebars-template'>
```
Comment (evilebottnawi) [inline tests/handlebars-basics/__snapshots__/jsfmt.spec.js.snap]: Problem what we change default `"` on `'`. We should keep `"` quotes for html like syntax (it is preferable developers).

Comment (dcyriller) [reply tests/handlebars-basics/__snapshots__/jsfmt.spec.js.snap]: No, no, I don't change the default quote in this PR. In that specific test, the `singleQuote` option is true.

File: tests/handlebars-attr-node/brackets.hbs [hunk 3]
```diff
@@ -0,0 +1,2 @@
+<div title='My "title"'></div>
+<div title="My other 'title'"></div>
```
Comment (evilebottnawi) [inline tests/handlebars-attr-node/brackets.hbs]: Let's add here:
- `<div title='My title'></div>`
- `<div title="My title"></div>`

for consistently testing and we can merge :+1:

Comment (dcyriller) [reply tests/handlebars-attr-node/brackets.hbs]: 👍
I've added these two cases in a last commit.

**Pull request merged**